src/wallet/rpcnames.cpp

```diff
@@ -829,6 +831,167 @@ name_update ()
 
 /* ************************************************************************** */
 
+RPCHelpMan
+queuerawtransaction ()
+{
+  return RPCHelpMan ("queuerawtransaction",
+      "\nQueue a transaction for future broadcast.",
+      {
+          {"hexstring", RPCArg::Type::STR, RPCArg::Optional::NO, "The hex string of the raw transaction"},
+      },
+      RPCResult {RPCResult::Type::STR_HEX, "", "the transaction ID"},
+      RPCExamples {
+          HelpExampleCli("queuerawtransaction", "txhex") +
+          HelpExampleRpc("queuerawtransaction", "txhex")
+      },
+      [&] (const RPCHelpMan& self, const JSONRPCRequest& request) -> UniValue
+{
+  std::shared_ptr<CWallet> const wallet = GetWalletForJSONRPCRequest (request);
+  if (!wallet) return NullUniValue;
+
+  RPCTypeCheck (request.params,
+                {UniValue::VSTR});
+
+  // parse transaction from parameter
+  CMutableTransaction mtxParsed;
+  if (!DecodeHexTx(mtxParsed, request.params[0].get_str(), true, true))
+    throw JSONRPCError(RPC_DESERIALIZATION_ERROR, "TX decode failed");
+  CTransactionRef txParsed(MakeTransactionRef(mtxParsed));
+  const uint256& hashTx = txParsed->GetHash();
+
+  // Validate transaction
+  NodeContext& node = EnsureAnyNodeContext(request.context);
+  ChainstateManager& chainman = EnsureChainman(node);
+  CTxMemPool& mempool = EnsureMemPool(node);
+
+  {
+    LOCK (cs_main);
+    // Check validity
+    MempoolAcceptResult result = AcceptToMemoryPool(chainman.ActiveChainstate(), mempool, txParsed,
+      /* bypass_limits */ false, /* test_accept */ true);
+    // If it can be broadcast immediately, do that and return early.
+    if (result.m_result_type == MempoolAcceptResult::ResultType::VALID)
+    {
+      std::string unused_err_string;
+      // Don't check max fee.
+      const TransactionError err = BroadcastTransaction(node, txParsed, unused_err_string,
+        /* max_tx_fee */ 0, /* relay */ true, /* wait_callback */ false);
+      assert(err == TransactionError::OK);
```

Comment:
The `assert` makes me a bit uneasy in general, as that could lead to potential for crashing the node.  However, looking at the code, it seems that it will indeed be guaranteed to go through at least with the current implementation of `BroadcastTransaction`, so I think it is fine.

What do you think of changing it to just queueing the transaction nevertheless, just in case?  I'm fine with either option, just asking about your opinion on this.